src/audio/pipeline/pipeline-params.c

```diff
@@ -80,33 +80,15 @@ static int pipeline_comp_params(struct comp_dev *current,
 		.skip_incomplete = true,
 	};
 	int stream_direction = ppl_data->params->params.direction;
-	int end_type;
 	int err;
 
 	pipe_dbg(current->pipeline, "pipeline_comp_params(), current->comp.id = %u, dir = %u",
 		 dev_comp_id(current), dir);
 
-	if (!comp_is_single_pipeline(current, ppl_data->start)) {
-		/* If pipeline connected to the starting one is in improper
-		 * direction (CAPTURE towards DAI, PLAYBACK towards HOST),
-		 * stop propagation of parameters not to override their config.
-		 * Direction param of the pipeline can not be trusted at this
-		 * point, as it might not be configured yet, hence checking
-		 * for endpoint component type.
-		 */
-		end_type = comp_get_endpoint_type(current->pipeline->sink_comp);
-		if (stream_direction == SOF_IPC_STREAM_PLAYBACK) {
-			if (end_type == COMP_ENDPOINT_HOST ||
-			    end_type == COMP_ENDPOINT_NODE)
-				return 0;
-		}
-
-		if (stream_direction == SOF_IPC_STREAM_CAPTURE) {
-			if (end_type == COMP_ENDPOINT_DAI ||
-			    end_type == COMP_ENDPOINT_NODE)
-				return 0;
-		}
-	}
+	/* Don't propagate to pipelines in the opposite direction */
```

Comment:
what is the purpose of this change?

Reply:
Pure cleanup.  I had some pipeline changes in an earlier version that got moved into the component's trigger handler, and this made those easier.

It's split out in #8620 if you want to register complaints there.  I'll remove it from this PR next spin.